jfxtras-agenda/src/main/java/jfxtras/internal/scene/control/skin/agenda/base24hour/AppointmentAbstractPane.java

```diff
@@ -299,15 +299,17 @@ private void handleDrag(Agenda.Appointment appointment, LocalDateTime dragPickup
 		  || (dragPickupInDayHeader && dragDropInDayHeader)
 		) {				
 			// simply add the duration
-			Duration duration = Duration.between(dragPickupDateTime, dragDropDateTime);
+            boolean changed = false;
+			Duration duration = Duration.between(dragPickupDateTime, dragDropDateTime);
 			if (appointment.getStartLocalDateTime() != null) {
 				appointment.setStartLocalDateTime( appointment.getStartLocalDateTime().plus(duration) );
-				layoutHelp.callAppointmentChangedCallback(appointment);
+                changed = true;
 			}
 			if (appointment.getEndLocalDateTime() != null) {
-				appointment.setEndLocalDateTime( appointment.getEndLocalDateTime().plus(duration) );
-				layoutHelp.callAppointmentChangedCallback(appointment);
-			}
+				appointment.setEndLocalDateTime( appointment.getEndLocalDateTime().plus(duration) );
+				changed = true;
+			}
+			if (changed) layoutHelp.callAppointmentChangedCallback(appointment);

```

Comment:
A valid appointment can't start after it ends.  It also can't end before it starts.  By calling callAppointmentChangedCallback on changes to start and then end separately can result in an invalid appointment
before the second change is recorded.  This problem is avoided by updating both changes in one call.